src/python/pants/backend/adhoc/target_types.py

```diff
@@ -0,0 +1,248 @@
+    alias = "extra_env_vars"
+    help = help_text(
+        """
+        Additional environment variables to provide to the process.
+
+        Entries are strings in the form `ENV_VAR=value` to use explicitly; or just
+        `ENV_VAR` to copy the value of a variable in Pants's own environment.
+        """
+    )
+
+
+class AdhocToolLogOutputField(BoolField):
+    alias = "log_output"
+    default = False
+    help = "Set to true if you want the output logged to the console."
+
+
+class AdhocToolWorkdirField(StringField):
+    alias = "workdir"
+    default = "."
+    help = help_text(
+        """
+        Sets the working directory for the process.
+
+        Values are relative to the build root, except in the following cases:
+
+        * `.` specifies the location of the `BUILD` file.
+        * Values beginning with `./` are relative to the location of the `BUILD` file.
+        * `/` or the empty string specifies the build root.
+        * Values beginning with `/` are also relative to the build root.
+        """
+    )
+
+
+class AdhocToolOutputRootDirField(StringField):
+    alias = "root_output_directory"
+    default = "/"
+    help = help_text(
+        """Adjusts the location of files output by this target, when consumed as a dependency.
+
+        Values are relative to the build root, except in the following cases:
+
+        * `.` specifies the location of the `BUILD` file.
+        * Values beginning with `./` are relative to the location of the `BUILD` file.
+        * `/` or the empty string specifies the build root.
+        * Values beginning with `/` are also relative to the build root.
+        """
+    )
+
+
+class AdhocToolTarget(Target):
+    alias = "adhoc_tool"
+    deprecated_alias = "experimental_run_in_sandbox"
+    deprecated_alias_removal_version = "2.17.0.dev0"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        AdhocToolRunnableField,
+        AdhocToolArgumentsField,
+        AdhocToolExecutionDependenciesField,
+        AdhocToolOutputDependenciesField,
+        AdhocToolLogOutputField,
+        AdhocToolOutputFilesField,
+        AdhocToolOutputDirectoriesField,
+        AdhocToolSourcesField,
+        AdhocToolTimeoutField,
+        AdhocToolExtraEnvVarsField,
+        AdhocToolWorkdirField,
+        AdhocToolOutputRootDirField,
+        AdhocToolStdoutFilenameField,
+        AdhocToolStderrFilenameField,
+        EnvironmentField,
+    )
+    help = help_text(
+        lambda: f"""
+        Execute any runnable target for its side effects.
+
+        Example BUILD file:
+
+            {AdhocToolTarget.alias}(
+                {AdhocToolRunnableField.alias}=":python_source",
+                {AdhocToolArgumentsField.alias}=[""],
+                {AdhocToolExecutionDependenciesField.alias}=[":scripts"],
+                {AdhocToolOutputDirectoriesField.alias}=["logs/my-script.log"],
+                {AdhocToolOutputFilesField.alias}=["results/"],
```

Comment:
When looking at the rendered docs it was much easier to spot that the example field values here are swapped (files vs directories). 👀